src/mmw/js/src/analyze/views.js

```diff
@@ -452,6 +453,87 @@ var ChartView = Marionette.ItemView.extend({
     }
 });
 
+var AnalyzeResultView = Marionette.LayoutView.extend({
+    template: analyzeResultsTmpl,
+    regions: {
+        chartRegion: '.chart-region',
+        tableRegion: '.table-region'
+    }
+});
+
+var LandResultView  = AnalyzeResultView.extend({
+    onShow: function() {
+        var categories = this.model.get('categories'),
+            largestArea = _.max(_.pluck(categories, 'area')),
+            units = utils.magnitudeOfArea(largestArea),
+            census = new coreModels.LandUseCensusCollection(categories);
+
+        this.chartRegion.show(new ChartView({
+            model: this.model,
+            collection: census
+        }));
+
+        this.tableRegion.show(new TableView({
+            units: units,
+            collection: census
+        }));
+    }
+});
+
+var SoilResultView  = AnalyzeResultView.extend({
+    onShow: function() {
+        var categories = this.model.get('categories'),
+            largestArea = _.max(_.pluck(categories, 'area')),
+            units = utils.magnitudeOfArea(largestArea),
+            census = new coreModels.SoilCensusCollection(categories);
+
+        this.chartRegion.show(new ChartView({
+            model: this.model,
+            collection: census
+        }));
+
+        this.tableRegion.show(new TableView({
+            units: units,
+            collection: census
+        }));
+    }
+});
+
+var AnimalsResultView = AnalyzeResultView.extend({
+    onShow: function() {
+        var categories = this.model.get('categories'),
+            largestArea = _.max(_.pluck(categories, 'area')),
+            units = utils.magnitudeOfArea(largestArea),
+            census = new coreModels.AnimalCensusCollection(categories);
+
+        this.tableRegion.show(new AnimalTableView({
+            units: units,
+            collection: census
+        }));
+    }
+});
+
+var PointSourceResultView = AnalyzeResultView.extend({
+    onShow: function() {
+        var categories = this.model.get('categories'),
```

Comment:
Consider extracting out a single function from the `onShow` functions for the four `ResultView` classes since they only differ in the `CensusCollection` and `TableView` classes that are used.